apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -0,0 +1,159 @@
+import jwt from 'jsonwebtoken';
+import { HttpService } from '@nestjs/axios';
+import JwksRsa from 'jwks-rsa';
+import QueryString from 'qs';
+import { lastValueFrom } from 'rxjs';
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { System, User } from '@shared/domain';
+import { Inject } from '@nestjs/common';
+import { Logger } from '@src/core/logger';
+import { SymetricKeyEncryptionService } from '@shared/infra/encryption';
+import { SystemRepo, UserRepo } from '@shared/repo';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { TokenRequestPayloadMapper } from '../mapper/token-request-payload.mapper';
+import { TokenRequestPayload } from '../controller/dto/token-request.payload';
+import { OAuthSSOError } from '../error/oauth-sso.error';
+import { OauthTokenResponse } from '../controller/dto/oauth-token.response';
+import { FeathersJwtProvider } from '../../authorization';
+import { IservOAuthService } from './iserv-oauth.service';
+import { IJWT } from '../interface/jwt.base.interface';
+import { OAuthResponse } from './dto/oauth.response';
+import { AuthorizationParams } from '../controller/dto/authorization.params';
+
+@Injectable()
+export class OAuthService {
+	constructor(
+		private readonly userRepo: UserRepo,
+		private readonly systemRepo: SystemRepo,
+		private readonly jwtService: FeathersJwtProvider,
+		private httpService: HttpService,
+		@Inject('OAuthEncryptionService') private readonly oAuthEncryptionService: SymetricKeyEncryptionService,
```

Comment:
Why you use an explicit inject?

Reply:
The OAuthEncryptionService is a custumized form of the class SymmetricEncryptionService. The normal case uses a different key. That is why we have altered the key input in ouath.module.ts and generated our own encryption service which we called the way you see it here. Because this is now customized we cannot use a simple import but also have to inject it here.